accord-core/src/main/java/accord/api/Read.java

```diff
@@ -29,5 +35,42 @@
 public interface Read
 {
     Keys keys();
-    Data read(Key key, Timestamp executeAt, DataStore store);
+    Future<Data> read(Key key, CommandStore commandStore, Timestamp executeAt, DataStore store);
+
+    class ReadFuture extends AsyncPromise<Data> implements BiConsumer<Data, Throwable>
```

Comment:
Should this extend `SyncPromise` since it appears to be synchronous? Or perhaps `AbstractFuture` with no synchronisation (since we should probably be invoking callbacks on the home shard `CommandStore`)

I think it would also be preferable not to have this defined in the interface.